Mathlib/Order/KrullDimension.lean

```diff
@@ -160,12 +220,23 @@ lemma height_eq_index_of_length_eq_height_last {p : LTSeries α} (h : p.length =
   norm_cast at *
   omega
 
+/--
+In a maximally long series, i.e one as long as the coheight of the first element, the coheight of
+each element is its reverse index in the series.
+-/
+lemma coheight_eq_index_of_length_eq_head_coheight {p : LTSeries α} (h : p.length = coheight p.head)
+    (i : Fin (p.length + 1)) : coheight (p i) = i.rev := by
+  simpa using height_eq_index_of_length_eq_height_last (α := αᵒᵈ) (p := p.reverse) (by simpa) i.rev
+
 lemma height_mono : Monotone (α := α) height :=
   fun _ _ hab ↦ biSup_mono (fun _ hla => hla.trans hab)
 
 @[gcongr] protected lemma _root_.GCongr.height_le_height (a b : α) (hab : a ≤ b) :
     height a ≤ height b := height_mono hab
 
+lemma coheight_mono : Antitone (α := α) coheight :=
+  fun _ _ hxy => height_mono (α := αᵒᵈ) hxy
```

Comment:
```suggestion
  height_mono.dual_left
```
might work

Reply:
Unfortunately doesn’t:
```
synthesized type class instance is not definitionally equal to expression inferred by typing rules, synthesized
  inst✝
inferred
  OrderDual.instPreorder α
```
And instantiating `α` gives
```
type mismatch
  Monotone.dual_left height_mono
has type
  Antitone (height ∘ ⇑OrderDual.ofDual) : Prop
but is expected to have type
  Antitone coheight : Prop
```
which doesn’t seem to line up, not sure why exactly.

Reply:
> And instantiating `α` gives

Do you mean
```
height_mono (α := α).dual_left
```
? because the right thing would be
```
height_mono (α := αᵒᵈ).dual_left
```

Reply:
Ah, yes, that may explain my failure. I'll try!